tools/libpico/CMakeLists.txt

```diff
@@ -1,216 +1,237 @@
-    pico_standard_link
-    pico_stdlib
-    pico_unique_id
-    pico_util
-    tinyusb
-    tinyusb_device_unmarked
-)
-
-add_library(ipv4 STATIC)
-target_compile_definitions(ipv4 PUBLIC
-    LWIP_IPV6=0
-)
-
-add_library(ipv4-big STATIC)
-target_compile_definitions(ipv4-big PUBLIC
-    __LWIP_MEMMULT=2
-    LWIP_IPV6=0
-)
-
-set(picow_link_libraries
-    common
-    pico_cyw43_driver
-    cyw43_driver
-    cyw43_driver_picow
-    pico_async_context
-    pico_cyw43_driver
-    pico_cyw43_arch
-    pico_cyw43_arch_threadsafe_background
-    pico_lwip
-    pico_lwip_nosys
-    pico_lwip_sntp
-    pico_stdlib
-)
-
-target_link_libraries(ipv4
-    ${picow_link_libraries}
-)
-
-target_link_libraries(ipv4-big
-    ${picow_link_libraries}
-)
-
-add_library(ipv4-ipv6 STATIC)
-target_compile_definitions(ipv4-ipv6 PUBLIC
-    LWIP_IPV6=1
-)
-
-add_library(ipv4-ipv6-big STATIC)
-target_compile_definitions(ipv4-ipv6-big PUBLIC
-    __LWIP_MEMMULT=2
-    LWIP_IPV6=1
-)
-
-target_link_libraries(ipv4-ipv6
-    ${picow_link_libraries}
-)
-
-target_link_libraries(ipv4-ipv6-big
-    ${picow_link_libraries}
-)
-
-add_library(ipv4-bt STATIC)
-target_compile_definitions(ipv4-bt PUBLIC
-    LWIP_IPV6=0
-)
-
-add_library(ipv4-bt-big STATIC)
-target_compile_definitions(ipv4-bt PUBLIC
-    __LWIP_MEMMULT=2
-    LWIP_IPV6=0
-)
-
-set(picow_bt_link_libraries
-    pico_btstack_cyw43
-    pico_btstack_ble
-    pico_btstack_classic
-    pico_btstack_sbc_encoder
-    pico_btstack_sbc_decoder
-)
-
-target_link_libraries(ipv4-bt
-    ${picow_link_libraries}
-    ${picow_bt_link_libraries}
-)
-
-target_link_libraries(ipv4-bt-big
-    ${picow_link_libraries}
-    ${picow_bt_link_libraries}
-)
-
-add_library(ipv4-ipv6-bt STATIC)
-target_compile_definitions(ipv4-ipv6-bt PUBLIC
-    LWIP_IPV6=1
-)
-
-add_library(ipv4-ipv6-bt-big STATIC)
-target_compile_definitions(ipv4-ipv6-bt-big PUBLIC
-    __LWIP_MEMMULT=2
-    LWIP_IPV6=1
-)
-
-target_link_libraries(ipv4-ipv6-bt
-    ${picow_link_libraries}
-    ${picow_bt_link_libraries}
-)
-
-target_link_libraries(ipv4-ipv6-bt-big
-    ${picow_link_libraries}
-    ${picow_bt_link_libraries}
-)
-
-
-foreach(tgt pico ipv4 ipv4-ipv6 ipv4-bt ipv4-ipv6-bt ipv4-big ipv4-ipv6-big ipv4-bt-big ipv4-ipv6-bt-big)
-    add_custom_command(TARGET ${tgt} POST_BUILD
-        COMMAND ar d lib${tgt}.a stdio.c.obj stdio_uart.c.obj stdio_usb.c.obj stdio_usb_descriptors.c.obj pico_malloc.c.obj newlib_interface.c.obj
-        COMMAND ar d lib${tgt}.a btstack_flash_bank.c.obj  # Need to override with our own implementation
-        COMMAND cp lib${tgt}.a ../../../lib/.
+foreach(cpu "rp2040" "rp2350") 
+    message("Building for CPU ${cpu}")
+
+    if (${cpu} MATCHES "rp2040")
+        # Enable PicoW driver support.  Compatible with standard Pico
+        set(PICO_BOARD pico_w)
+        set(PICO_PLATFORM rp2040)
+        set(PICO_CYW43_SUPPORTED 1)
+    elseif(${cpu} MATCHES "rp2350")
+        set(PICO_BOARD pico_w)
+        set(PICO_PLATFORM rp2040)
```

Comment:
This should probably be rp2350

Reply:
Yeah, that's a doozy.  Definitely cut-n-paste error on my part and it would bork all the RP2350 libs!  It would still link in the 2040 register map/etc. in the SDK library so things like clock startup would be...interesting. :laughing: